### PR TITLE
Update JVM Heap Size section

### PR DESCRIPTION
**Pull request by alaudazzi**
This PR updates the JVM Heap Size instructions, based on the recent ES automatic settings.

Fixes: #4089

### CODE REVIEW

**Comment by alaudazzi** [comment]
@elasticmachine update branch

**Comment by alaudazzi** [comment]
@elasticmachine run elasticsearch-ci/docs

File: docs/orchestrating-elastic-stack-applications/elasticsearch/jvm-heap-size.asciidoc [hunk 1]
```diff
@@ -8,8 +8,12 @@ endif::[]
 [id="{p}-{page_id}"]
 = JVM heap size
 
-To change the heap size of Elasticsearch, set the `ES_JAVA_OPTS` environment variable in the `podTemplate`. It is also highly recommended to set the resource `requests` and `limits` at the same time to ensure that the pod gets enough resources allocated within the Kubernetes cluster. See <<{p}-compute-resources-elasticsearch>> for an example and more information.
+These instructions apply if you are using:
+- Versions of Elasticsearch before 7.11
+- Elasticsearch 7.11 or later, and you want to change the default heap size. 
 
-If `ES_JAVA_OPTS` is not defined, the Elasticsearch default heap size of 1Gi will be in effect.
+To change the heap size of Elasticsearch, set the `ES_JAVA_OPTS` environment variable in the `podTemplate`. It is also highly recommended to set the resource `requests` and `limits` at the same time to ensure that the Pod gets enough resources allocated within the Kubernetes cluster. See <<{p}-compute-resources-elasticsearch>> for an example and more information.
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/jvm-heap-size.asciidoc:15]: I think the linked document (https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-managing-compute-resources.html#k8s-compute-resources-elasticsearch) also needs an update to explain the changes in 7.11 and perhaps a link to the [Elasticsearch sizing guide](https://www.elastic.co/guide/en/elasticsearch/reference/7.11/advanced-configuration.html).

File: docs/orchestrating-elastic-stack-applications/managing-compute-resources.asciidoc [hunk 2]
```diff
@@ -19,8 +19,7 @@ You can set compute resource constraints in the `podTemplate` of objects managed
 [id="{p}-compute-resources-elasticsearch"]
 === Set compute resources for Elasticsearch
 
-For Elasticsearch objects, make sure to consider the heap size when you set resource requirements.
-A good rule of thumb is to size it to half the size of RAM allocated to the Pod.
+Staring with Elasticsearch 7.11 and later, the JVM heap size is automatically adjusted. For older Elasticsearch versions, you need to follow the instructions on <<{p}-jvm-heap-size,how to configure the JVM heap size>>.
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/managing-compute-resources.asciidoc:22]:
```suggestion
NOTE: Staring with Elasticsearch 7.11, unless manually overridden, heap size is automatically calculated based on the node roles and the available memory. In Kubernetes, the amount of memory available to an Elasticsearch node is determined by the `limits` defined for that container. See <<{p}-jvm-heap-size>> for more information.
```

**Review by charith-elastic — COMMENTED** [review]
I did some tests to figure out how Elasticsearch 7.11 works on Kubernetes and added a few suggested additions to the text.

**Review by charith-elastic — APPROVED** [review]
LGTM

**Pull request merged**